File: springapp/src/main/java/com/example/springapp/model/Employee.java
package com.example.springapp.model;

import jakarta.persistence.*;
import lombok.Data;
import java.time.LocalDate;

@Data
@Entity
@Table(name = "employees")
public class Employee {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(nullable = false)
    private String firstName;

    @Column(nullable = false)
    private String lastName;

    @Column(unique = true, nullable = false)
    private String email;

    private String phoneNumber;
    private String designation;
    private String department;
    private String position;
    private LocalDate hireDate;
    private Double salary;
    private String status; // ACTIVE, INACTIVE, TERMINATED
    private String address;
    private String emergencyContact;
    private String emergencyPhone;

    // ✅ One-to-One with UserAccount
    @OneToOne(mappedBy = "employee", cascade = CascadeType.ALL)
    private UserAccount userAccount;
}

File: angularapp/src/models/employee.ts
export interface Employee {
  id?: number | Long; // Support both number and Long
  firstName: string;
  lastName: string;
  email: string;
  phoneNumber?: string;
  designation?: string;
  department?: string;
  position?: string;
  hireDate?: string; // ISO date string
  salary?: number;
  status?: string; // ACTIVE, INACTIVE, TERMINATED
  address?: string;
  emergencyContact?: string;
  emergencyPhone?: string;
  
  // Additional fields to match backend
  fullName?: string; // Derived from firstName + lastName
}

// Type alias for compatibility
type Long = number;
